src/remote.c

```diff
@@ -299,47 +321,60 @@ int git_remote_create(git_remote **out, git_repository *repo, const char *name,
 {
 	git_buf buf = GIT_BUF_INIT;
 	int error;
+	git_remote_create_options opts = GIT_REMOTE_CREATE_OPTIONS_INIT;
 
-	if (default_fetchspec_for_name(&buf, name) < 0)
-		return -1;
+	/* Those 2 tests are duplicated here because of backward-compatibility */
+	if ((error = ensure_remote_name_is_valid(name)) < 0)
+		return error;
+
+	if (canonicalize_url(&buf, url) < 0)
+		return GIT_ERROR;
```

Comment:
What kind of backwards compatibility do you refer to? Do callers expect a specific error code here?

Reply:
I found out while writing [this test](https://github.com/libgit2/libgit2/pull/4667/commits/eb725118f18a92cfcf05de58f4ff90139af9c82f#diff-0f6c0f30ddbd94614e1f4f4887955556R93) that [this](https://github.com/libgit2/libgit2/blob/master/src/remote.c#L322) was "misbehaving", in that it never returns what `create_internal` says when it fails. It depends on what kind of backward compatibility we want (I realize I may be too zealous, but maybe this should be discussed separately)…

The `ensure_remote_name_is_valid` is duplicated in case name is NULL (a NULL name in the `with_opts` version would cause an anonymous remote to be created instead, ie. it's not a failure anymore). An alternative would be to assert on the mandatory parameters to be correct…

Reply:
Yeah. I guess we should fix this, but that can be done in a later PR if you wish.